### PR TITLE
tests(aes-gcm): add  edge cases  and negative tests cases 

### PR DESCRIPTION
**Pull request by meilu-git**
tests(aes-gcm): add  edge cases  and negative tests cases

### CODE REVIEW

File: api/tests/src/algo/aes/gcm_tests.rs [hunk 1]
```diff
@@ -440,3 +447,596 @@ fn test_gcm_wrong_aad_fails(session: HsmSession) {
...
+    let enc_algo = new_gcm_encrypt_algo(&iv, None);
+    let mut ctx = enc_algo.encrypt_init(key).unwrap();
+
+    let _ = ctx.finish_vec().unwrap();
+
+    let bytes = ctx.update_vec(&plaintext).unwrap();
+    assert!(
+        bytes.is_empty(),
+        "update() after finish() should be a no-op for GCM"
+    );
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply api/tests/src/algo/aes/gcm_tests.rs]: The current API allows update after finish and only guarantees no immediate output for GCM update(). Test reflects current semantics.

**Review by rajesh-gali — APPROVED** [review]
